ga4gh/converters.py

```diff
@@ -6,35 +6,236 @@
+        else:
+            fileString = "-"
+        alignmentFile = pysam.AlignmentFile(
+            fileString, flags, header=header.toDict())
+        iterator = self._httpClient.searchReads(self._searchReadsRequest)
+        for read in iterator:
+            alignedSegment = SamLine.toAlignedSegment(read, targetIds)
+            alignmentFile.write(alignedSegment)
+        alignmentFile.close()
+
+    def _getHeader(self):
+        header = SamHeader()
+        return header
+
+    def _getTargetIds(self, header):
+        # this seems to be how pysam sets the target ids
+        headerDict = header.toDict()
+        targetIds = collections.defaultdict(int)
+        targetId = 0
+        if 'SQ' in headerDict:
+            headerLines = headerDict['SQ']
+            for headerLine in headerLines:
+                refName = headerLine['SN']
+                targetIds[refName] = targetId
+                targetId += 1
+        return targetIds
+
+
+class SamHeader(object):
+    """
+    The header of a SAM file
+    """
+    def __init__(self):
+        # TODO where to get actual values for header?
+        # need some kind of getReadGroup(readGroupId) method in protocol
+        # just add these dummy lines for now
+        self.header = {
+            'HD': {'VN': '1.0'},
+            'SQ': [
+                {'LN': 1575, 'SN': 'chr1'},
+                {'LN': 1584, 'SN': 'chr2'},
+            ],
+        }
+
+    def toDict(self):
+        return self.header
+
+
+class SamLine(object):
+    """
+    Methods for processing a line in a SAM file
+    """
+    # see http://pysam.readthedocs.org/en/latest/api.html
+    # #pysam.AlignedSegment.cigartuples
+    cigarMap = {
+        protocol.GACigarOperation.ALIGNMENT_MATCH: 0,
+        protocol.GACigarOperation.INSERT: 1,
+        protocol.GACigarOperation.DELETE: 2,
+        protocol.GACigarOperation.SKIP: 3,
+        protocol.GACigarOperation.CLIP_SOFT: 4,
+        protocol.GACigarOperation.CLIP_HARD: 5,
+        protocol.GACigarOperation.PAD: 6,
+        protocol.GACigarOperation.SEQUENCE_MATCH: 7,
+        protocol.GACigarOperation.SEQUENCE_MISMATCH: 8,
+    }
+
+    # see tables in SAM spec, section 1.5
+    _tagReservedFieldPrefixes = set(["X", "Y", "Z", ])
+    _tagIntegerFields = set([
+        "AM", "AS", "CM", "CP", "FI", "H0", "H1", "H2", "HI", "IH", "MQ",
+        "NH", "NM", "OP", "PQ", "SM", "TC", "UQ", ])
+    _tagStringFields = set([
+        "BC", "BQ", "CC", "CO", "CQ", "CS", "CT", "E2", "FS", "LB", "MC",
+        "MD", "OQ", "OC", "PG", "PT", "PU", "QT", "Q2", "R2", "RG", "RT",
+        "SA", "U2", ])
+    _tagIntegerArrayFields = set(["FZ", ])
+
+    def __init__(self):
+        raise SamException("SamLine can't be instantiated")
+
+    @classmethod
+    def toAlignedSegment(cls, read, targetIds):
+        alignedSegment = pysam.AlignedSegment()
+        # QNAME
+        alignedSegment.query_name = \
+            read.fragmentName.encode('utf8')
```

Comment:
Make a class variable `_encoding` and set this to 'utf8'.  I find these line breaks really ugly and difficult to read also. Can we use a short variable name here? What about `ret`? I.e.
```
ret = pysam.AlignedSegment()
ret.query_name = read.fragmentName.encode(self._encoding)
#etc
```
Since the method name is `toAlignedSegment` this should be abundantly clear?